packages/common/src/components/PkgGraph.tsx

```diff
@@ -0,0 +1,87 @@
+import {
+  ReactWidget
+} from '@jupyterlab/apputils';
+
+import { Graph, GraphData, GraphNode, GraphLink } from "react-d3-graph";
+
+import * as React from 'react';
+
+import { Conda } from '../tokens';
+
+export class PkgGraph extends ReactWidget {
+  constructor(pkgManager: Conda.IPackageManager, pkg: string) {
+    super();
+    this._package = pkg;
+    this._pkgManager = pkgManager;
+    this._loading = true;
+
+    // create a network
+    this._data = { nodes: [], links: [] };
+    if (this._package) this._updatePackages();
+  }
+
+  public showGraph(pkg: string): void {
+    this._package = pkg;
+    this._updatePackages();
+  }
+
+  private async _updatePackages(): Promise<void> {
+    this._loading = true;
+    this.update();
+
+    this._pkgManager.getDependencies(this._package, true)
+    .then( available => {
```

Comment:
```suggestion
    const available =  await this._pkgManager.getDependencies(this._package, true);
```

The code uses as much as possible `async/await` syntax rather than `then`. Please try to use that syntax too.